src/stacked_borrows.rs

```diff
@@ -226,32 +226,33 @@ impl<'tcx> Stack {
     /// Find the item granting the given kind of access to the given tag, and return where
     /// it is on the stack.
     fn find_granting(&self, access: AccessKind, tag: Tag) -> Option<usize> {
-        self.borrows.iter()
+        self.borrows
+            .iter()
             .enumerate() // we also need to know *where* in the stack
             .rev() // search top-to-bottom
             // Return permission of first item that grants access.
             // We require a permission with the right tag, ensuring U3 and F3.
-            .find_map(|(idx, item)|
+            .find_map(|(idx, item)| {
                 if tag == item.tag && item.perm.grants(access) {
                     Some(idx)
                 } else {
                     None
                 }
-            )
+            })
     }
 
     /// Find the first write-incompatible item above the given one --
     /// i.e, find the height to which the stack will be truncated when writing to `granting`.
     fn find_first_write_incompatible(&self, granting: usize) -> usize {
         let perm = self.borrows[granting].perm;
         match perm {
-            Permission::SharedReadOnly =>
-                bug!("Cannot use SharedReadOnly for writing"),
-            Permission::Disabled =>
-                bug!("Cannot use Disabled for anything"),
+            Permission::SharedReadOnly => bug!("Cannot use SharedReadOnly for writing"),
+            Permission::Disabled => bug!("Cannot use Disabled for anything"),
             Permission::Unique =>
-                // On a write, everything above us is incompatible.
-                granting + 1,
+            // On a write, everything above us is incompatible.
+            {
+                granting + 1
```

Comment:
If rustfmt decides to mess up the formatting here, at least code and comment should stay together...